libs/@guardian/identity-auth/src/identityAuth.ts

```diff
@@ -52,6 +55,10 @@ export class IdentityAuth<
 			keysUrl: `${this.#options.issuer}/v1/keys`,
 		};
 
+		// before doing anything else, we check if the user's session should be refreshed based on the idCookieSessionRefresh option
+		const profileUrl = this.#options.issuer.split('/oauth2')[0] as ProfileUrl;
```

Comment:
Is this type assertion necessary? Could you have a proper handling of the case where the `profileUrl` is incorrect?

Reply:
I'll add a type guard for the URL, I used the assertion, as the type for the `issuer` always began with one of the `profile` urls, but when using `split` and getting the first element, typescript only thought it was a string, and not a specific type.

Reply:
Just the [`dotcom-rendering`’s `guard` implementation](https://github.com/guardian/dotcom-rendering/blob/21f02cdf5418b1d4472f4d64d8343a939f536a58/dotcom-rendering/src/lib/guard.ts) here in case…

Reply:
ooh

Reply:
updated to use your amazing type guard, have added you as the author for that commit!